sympy/solvers/solveset.py

```diff
@@ -1111,44 +1074,195 @@ def _is_exponential(f, symbol):
+
+    symbol : Symbol
+        The variable in which the equation is solved
+
+    Returns
+    =======
+
+    An improved equation containing a single instance of log.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, S, log, Eq
+    >>> from sympy.solvers.solveset import \
+    ... _solve_logarithm_reducible_to_single_instance as solve_log
+    >>> x = symbols('x')
+    >>> f = Eq(log(x - 3) + log(x + 3), 0)
+    >>> solve_log(f, x)
+    log((x - 3)*(x + 3))
+
+    * Proof of correctness
+
+    A logarithm is another way to write exponent and is defined by
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    When one side of the equation contains a single logarithm, the
+    equation can be solved by rewriting the equation as an equivalent
+    exponential equation as defined above. But if one side contains
+    more than one logarithm, we need to use the properties of logarithm
+    to condense it into a single logarithm.
+
+    Take for example
+
+    .. math:: log(2*x) - 15 = 0
+
+    contains single logarithm, therefore we can directly rewrite it to
+    exponential form as
+
+    .. math:: x = exp(15)/2
+
+    But if the equation has more than one logarithm as
+
+    .. math:: log(x - 3) + log(x + 3) = 0
+
+    we use log identities to convert into a reduced form
+
+    Using,
+
+    .. math:: log(a) + log(b) = log(a*b)
+
+    therefore the equation becomes,
+
+    .. math:: log((x - 3)*(x  + 3))
+
+    This equation contains one logarithm and can be solved by rewriting
+    to exponents.
+    """
+
+    lhs, rhs = f.lhs, f.rhs
+    new_lhs = logcombine(lhs, force=True)
+    new_f = new_lhs - rhs
+
+    return new_f
+
+
+def _is_logarithm_reducible_to_single_instance(f, symbol):
+    r"""
+    Helper to check whether an equation is logarithmic or not.
+
+    Logarithmic equations are the equations that contains
+    `log` terms which can be reduced to a single log term or
+    a constant using various logarithmic identities.
+
+    For example:
+
+    .. math:: log(x) + log(x - 4)
+
+    can be reduced to:
+
+    .. math:: log(x(x - 4))
+
+    Parameters
+    ==========
+
+    f : Expr
+        The equation to be checked as an instance of `Eq`
+
+    symbol : Symbol
+        The variable in which the equation is checked
+
+    Returns
+    =======
+
+    `True` if the equation is reducible otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, tan, log, Eq
+    >>> from sympy.solvers.solveset import \
+    ... _is_logarithm_reducible_to_single_instance as check
+    >>> x = symbols('x')
+    >>> check(Eq(log(x + 2) - log(x + 3), 0), x)
+    True
+    >>> check(Eq(tan(log(2*x)), 0), x)
+    False
+
+    * Philosophy behind the helper
+
+    The function uses `logcombine` to see whether the equation
+    gets reduced to a single instance of `log`.
+    """
+    lhs, rhs = f.lhs, f.rhs
+    new_lhs = logcombine(lhs, force=True)
+    if new_lhs != lhs:
```

Comment:
-1

This only tells you that some log terms combined, not that it was in the form of a sum of terms containing logs as factors, e.g.

```python
>>> logcombine(log(x)+log(y)+x, force=True)
x+log(xy)
```

Reply:
Ohkay! so I can have a check after this condition `if isinstance(new_lhs, log) or new_lhs.is_zero` `new_lhs.is_zero` is for cases like: `log(x**y) - y*log(x)`